driver/src/test/java/org/neo4j/driver/internal/connector/socket/ChunkedInputTest.java

```diff
@@ -150,4 +152,30 @@ public void canReadAllIntegerSizes() throws Exception
         assertEquals( 0,                 in.readInt() );
         assertEquals( Integer.MIN_VALUE, in.readInt() );
     }
-}
\ No newline at end of file
+
+    @Test
+    public void shouldNotReadMessageEndingWhenByteLeftInBuffer()
```

Comment:
In this case they are equivalent though - you only have a single line after the expect, so the test will fail if anything throws an exception before that, and it'll fail if that line throws any exception other than the one you expect. True though that there if there are cases where it can be unclear, try/finally is safer.